### PR TITLE
#35: Setup filling mask pipeline once

### LINKED ISSUE #35: Setup FillingMask Pipeline once for each model
# Background
- Current implementation setup filling-mask pipeline for each topk values of each model
- top_k parameter can be set in the __call__ method of the pipeline 
- Thus, setup call can be done once in each model

# Acceptance Criteria
- setup filling-mask pipeline once
- update tests accordingly

### PR DESCRIPTION
**Pull request by umitbuyuksahin**
Fixes #35

### CODE REVIEW

File: exasol_transformers_extension/udfs/models/filling_mask_udf.py [hunk 1]
```diff
@@ -136,65 +124,63 @@ def get_prediction(self, model_df: pd.DataFrame) -> pd.DataFrame:
...
 
         :return: Prepared dataframe including input data and predictions
         """
-        top_k = model_df['top_k'].iloc[0]
-
-        # Repeat each row consecutively as the number of labels. At the end,
-        # the dataframe is expanded from (m, n) to (m*top_k, n)
-        repeated_indexes = model_df.index.repeat(top_k)
+        n_topk_results = list(map(lambda x: x.shape[0], pred_df_list))
+        repeated_indexes = model_df.index.repeat(repeats=n_topk_results)
```
Comment (tkilias) [inline exasol_transformers_extension/udfs/models/filling_mask_udf.py:178]: N_topk_results looks like a list of tuples of count, value. Does repeat handle this.

Comment (umitbuyuksahin) [reply exasol_transformers_extension/udfs/models/filling_mask_udf.py:178]: yes, exactly

File: exasol_transformers_extension/udfs/models/question_answering_udf.py [hunk 2]
```diff
@@ -112,6 +110,8 @@ def load_models(self, model_name: str) -> None:
             device=self.device,
             framework="pt")
 
+        self.last_loaded_model = self.last_loaded_model.to(self.device)
```
[the remark this answers is not in the document]
Comment (umitbuyuksahin) [reply exasol_transformers_extension/udfs/models/question_answering_udf.py]: right, it is redundant, removed them

File: exasol_transformers_extension/udfs/models/filling_mask_udf.py [hunk 3]
```diff
@@ -134,65 +124,63 @@ def get_prediction(self, model_df: pd.DataFrame) -> pd.DataFrame:
...
-        text_data_with_valid_mask_token = \
-            [text_data.replace(self.mask_token,
-                               self.last_created_pipeline.tokenizer.mask_token)
-             for text_data in text_data_raw]
-        results = self.last_created_pipeline(text_data_with_valid_mask_token)
+        text_data_with_valid_mask_token = [
+            text_data.replace(
+                self.mask_token,
+                self.last_created_pipeline.tokenizer.mask_token
+            ) for text_data in text_data_raw]
```
Comment (tkilias) [inline exasol_transformers_extension/udfs/models/filling_mask_udf.py:146]: extract to function

File: exasol_transformers_extension/udfs/models/filling_mask_udf.py [hunk 4]
```diff
@@ -134,65 +124,63 @@ def get_prediction(self, model_df: pd.DataFrame) -> pd.DataFrame:
...
+        results_df_list = []
         for result in results:
-            filled_texts.append(result['sequence'])
-            scores.append(result['score'])
-
-        return scores, filled_texts
+            result_df = pd.DataFrame(result)
+            result_df = result_df[columns].rename(
+                columns={"sequence": "filled_text"})
+            results_df_list.append(result_df)
```
Comment (tkilias) [inline exasol_transformers_extension/udfs/models/filling_mask_udf.py:161]: extract to method

**Review by tkilias — CHANGES_REQUESTED** [review]

File: exasol_transformers_extension/udfs/models/filling_mask_udf.py (hunk 4 above)
Comment (umitbuyuksahin) [reply exasol_transformers_extension/udfs/models/filling_mask_udf.py:161]: These refactorings are actually part of the refactoring ticket #19  . All model udfs have these methods, somehow we collect them in a common class

File: exasol_transformers_extension/udfs/models/filling_mask_udf.py (hunk 3 above)
Comment (umitbuyuksahin) [reply exasol_transformers_extension/udfs/models/filling_mask_udf.py:146]: These refactorings are actually part of the refactoring ticket #19  . All model udfs have these methods, somehow we collect them in a common class

File: exasol_transformers_extension/udfs/models/filling_mask_udf.py (hunk 4 above)
Comment (tkilias) [reply exasol_transformers_extension/udfs/models/filling_mask_udf.py:161]: ok, agree

File: exasol_transformers_extension/udfs/models/filling_mask_udf.py (hunk 3 above)
Comment (tkilias) [reply exasol_transformers_extension/udfs/models/filling_mask_udf.py:146]: I think, this is specific to the filling mask task, so I think, it is a good idea to do it here

**Review by tkilias — APPROVED** [review]

**Pull request merged**